includes/api/ApiQueryRecentChanges.php

```diff
@@ -190,25 +190,25 @@ public function run( $resultPageSet = null ) {
 		if ( !is_null( $params['user'] ) ) {
 			// SUS-812: handle anon cases (IP address provided) and account names (user name provided)
 			if ( IP::isIPAddress( $params['user'] ) ) {
-				$this->addWhereFld( 'rc_user_text', $params['user'] );
-				$index['recentchanges'] = 'rc_user_text';
-			}
-			else {
+				$this->addWhereFld( 'rc_ip_bin', inet_pton( $params['user'] ) );
+				$index['recentchanges'] = 'rc_ip_bin';
+			} else {
 				$this->addWhereFld( 'rc_user', User::idFromName( $params['user'] ) );
 				$index['recentchanges'] = 'rc_user';
 			}
 		}
 
 		if ( !is_null( $params['excludeuser'] ) ) {
+			// TODO SUS-3079: verify if this is still the case
 			// We don't use the rc_user_text index here because
 			// * it would require us to sort by rc_user_text before rc_timestamp
 			// * the != condition doesn't throw out too many rows anyway
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this commented out code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)

[project-id]: wikia:app